code/datums/wire_panel.dm

```diff
@@ -0,0 +1,179 @@
+		WPANEL_INDICATOR(WIRE_CONTROL_TRANSMIT, "lime", "blink", "off"),
+		)
+
+/**
+ * Definition of a single wire. Automatically generated via `/datum/wirePanel/panelDefintion/New()`.
+ *
+ * Arguments:
+ * * `controls`: bitfield of `WIRE_CONTROL_*`. - Which control wire flags does this control
+ * * `wire_color`: string - *Named* colour to apply to this wire.
+ * * `to_fix`?: bitfield of `WIRE_ACT_*` - Which actions on this wire will fix the related controls
+ * * `to_break`?: bitfield of `WIRE_ACT_*` - Which actions on this wire will break the related controls
+ */
+/datum/wirePanel/wireDefintion
+	var/control_flags
+	var/wire_color_name
+	var/wire_color_value
+	var/actions_to_fix
+	var/actions_to_break
+
+/datum/wirePanel/wireDefintion/New(controls=0, wire_color="iris", to_fix=~WIRE_ACT_CUT, to_break=~WIRE_ACT_MEND)
+	. = ..()
+	src.control_flags = controls
+	src.wire_color_name = wire_color
+	src.actions_to_fix = to_fix
+	src.actions_to_break = to_break
+
+/// Fetches the RGB color value from the global `named_colors`
+/datum/wirePanel/wireDefintion/proc/cache_color()
+	if (!src.wire_color_value)
+		var/datum/named_color/C = get_color_by_name(src.wire_color_name)
+		if (istype(C))
+			src.wire_color_value = rgb(C.r, C.g, C.b)
+
+/***
+ * Definition of a wire panel
+ *
+ * Randomly links wire color to controls. Use as a static variable to share wire color-to-function mapping.
+ *
+ * Arguments:
+ * * `controls`: indexed list of `WIRE_CONTROL_*` flags - Each wire can handle one or more wire controls
+ * * `color_pool`: unordered list of global `named_colors` colors - pool to pull from for wire color. must be equal to or larger than `controls` length.
+ * * `custom_acts`?: unordered list of `/datum/wirePanel/wireActs` - what will fix/activate each wire control
+ * * `indicators`?: bitfield of `WIRE_CONTROL_*` - which indicators are tracked. defaults to all controls in `wire_control`. set to -1 for no indicators.
+ * * `light_map`?: `/datum/indicatorLights` - Custom defintion of indicator lights. defaults to directly referencing `/datum/indicatorLights/indicatorMap`
+ * * `keep_order`?: boolean - keeps the order of wire controls and colors (e.g. for debugging)
+ */
+/datum/wirePanel/panelDefintion
+	/// Ordered list of `wirePanel/wireDefintion`. Defintions for each wire
+	var/list/datum/wirePanel/wireDefintion/wire_definitions = list()
+	/// Ordered List of `wirePanel/indicatorDefintion. All available indicator lights for this defintion
+	var/list/datum/wirePanel/indicatorDefintion/indicator_lights = list()
+
+/datum/wirePanel/panelDefintion/New(
+	list/controls,
+	list/color_pool,
+	list/custom_acts,
+	indicators,
+	datum/wirePanel/indicatorMap/light_map,
+	keep_order=FALSE,
+	)
+	. = ..()
+	var/count = length(controls)
+	if (count > length(color_pool))
+		CRASH("Tried to create wirePanelDefinition with longer length `controls` than `color_pool`")
+	if (istype(light_map))
+		if (!istype(light_map, /datum/wirePanel/indicatorMap))
```

Comment:
this can never be true, `istype` with a single argument checks if the value is of the declared type of the var

Reply:
It should have two args here - `light_map` and `/datum/wirePanel/indicatorMap`. Am I misreading that?

Reply:
There are two `istype`s, the first one checks if `light_map` is its assigned type `/datum/wirePanel/indicatorMap`, and then the second one checks if it isn't, which can never be true if it passed the first one.

Reply:
gotcha, I was expecting the first one to just check if it existed at all. simplifying.

Reply:
[f01b378](https://github.com/goonstation/goonstation/pull/13624/commits/f01b37837f935e8cbf08260e6d8e87ba2ac75cbb)